### PR TITLE
mocktikv: change check conflict in mocktikv in one loop

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

- Do some refactor for the check conflict logic in `mocktikv`, using the same iterator sequentially
- Using one iterator and loop only one time to make it faster for ci. For point get, the old logic will use `getValue` another time to read the value, which may slow the check operation

The `tidb_ghpr_common_test` execution time for this pr
```
9971
—
PR #15171: mocktikv: change check conf... https://github.com/pingcap/tidb/pull/15171
3m 20s
a few seconds ago

9972
—
PR #15171: mocktikv: change check conf... https://github.com/pingcap/tidb/pull/15171
2m 31s
a minute ago


```
@glorv  has helped to validate this running another 5 times average in 3 minutes

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

**Comment by cfzjywxk** [comment]
@tiancaiamao  @coocood @jackysp  PTAL

File: store/mockstore/mocktikv/mvcc_leveldb.go [hunk 1]
```diff
@@ -686,67 +681,81 @@ func (mvcc *MVCCLevelDB) Prewrite(req *kvrpcpb.PrewriteRequest) []error {
...
 		}
 	}
 
-	if m.Assertion == kvrpcpb.Assertion_NotExist {
-		for ok {
+	needGetVal := getVal
+	needCheckAssertion := m.Assertion == kvrpcpb.Assertion_NotExist
+	needCheckRollback := true
+	var retVal []byte
+	// do the check or get operations using one iterator to make it ct faster
```
Comment (Deardrops) [inline store/mockstore/mocktikv/mvcc_leveldb.go]:
```suggestion
	// do the check or get operations within one iterator to make CI faster
```

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**